lhotse/bin/modes/cut.py

```diff
@@ -89,26 +95,43 @@ def random_overlayed(
 
 
 @cut.command()
-@click.argument('supervision_manifest', type=click.Path(exists=True, dir_okay=False))
-@click.argument('feature_manifest', type=click.Path(exists=True, dir_okay=False))
-@click.argument('output_dir', type=click.Path())
-def stereo_overlayed(
-        supervision_manifest: Pathlike,
-        feature_manifest: Pathlike,
-        output_dir: Pathlike,
+@click.argument('cut_manifests', nargs=-1, type=click.Path(exists=True, dir_okay=False))
+@click.argument('output_cut_manifest', type=click.Path())
+def mix_sequential(
+        cut_manifests: List[Pathlike],
+        output_cut_manifest: Pathlike
 ):
     """
-    Create a CutSet stored in OUTPUT_CUT_MANIFEST that contains supervision regions from SUPERVISION_MANIFEST
-    and features supplied by FEATURE_MANIFEST. It assumes that every recording has two channels, for which both
-    the supervisions and the features exist. It sums the features of both channels producing a set of mixed cuts.
+    Create a CutSet stored in OUTPUT_CUT_MANIFEST by iterating jointly over CUT_MANIFESTS and mixing the Cuts
+    on the same positions. E.g. the first output cut is created from the first cuts in each input manifest.
+    The mix is performed by summing the features from all Cuts.
+    If the CUT_MANIFESTS have different number of Cuts, the mixing ends when the shorter manifest is depleted.
     """
-    supervision_set = SupervisionSet.from_yaml(supervision_manifest)
-    feature_set = FeatureSet.from_yaml(feature_manifest)
+    cut_manifests = [CutSet.from_yaml(path) for path in cut_manifests]
+    mixed_cuts = (
+        reduce(lambda left_cut, right_cut: left_cut.overlay(right_cut), cuts[1:], cuts[0])
+        for cuts in zip(*cut_manifests)
+    )
+    mixed_cut_set = CutSet(cuts={cut.id: cut for cut in mixed_cuts})
+    mixed_cut_set.to_yaml(output_cut_manifest)
 
-    source_cut_set = make_cuts_from_supervisions(supervision_set=supervision_set, feature_set=feature_set)
-    mixed_cut_set = mix_stereo_cut_set(source_cut_set)
 
-    output_dir = Path(output_dir)
-    output_dir.mkdir(exist_ok=True, parents=True)
-    source_cut_set.to_yaml(output_dir / 'source_cuts.yml')
-    mixed_cut_set.to_yaml(output_dir / 'mixed_cuts.yml')
+@cut.command()
+@click.argument('cut_manifests', nargs=-1, type=click.Path(exists=True, dir_okay=False))
+@click.argument('output_cut_manifest', type=click.Path())
+def mix_by_recording_id(
+        cut_manifests: List[Pathlike],
+        output_cut_manifest: Pathlike
+):
+    """
+    Create a CutSet stored in OUTPUT_CUT_MANIFEST by matching the Cuts from CUT_MANIFESTS by their recording IDs
+    and mixing them together.
+    """
+    all_cuts = combine(*[CutSet.from_yaml(path) for path in cut_manifests])
+    recording_id_to_cut = groupby(lambda cut: cut.recording_id, all_cuts)
+    mixed_cuts = (
+        reduce(lambda left_cut, right_cut: left_cut.overlay(right_cut), cuts[1:], cuts[0])
```

Comment:
Your python can be on the advanced side, making it harder to understand than if it were simple... having trouble with this.

Reply:
.. and I don't understand why you do 
 cuts[1:], cuts[0]

Reply:
Would be nice to clarify that it's a shallow operation.

If I have any concerns about all this, it's that I feel the code might be becoming a little opaque.

Not sure how to fix that though.

Reply:
Alright, I'll either add a plenty of comments or re-write a bit.

Reply:
Done, I refactored this into a `mix_cuts` function, and inside I explained what this code does.